src/components/profile/ProfileCard.tsx

```diff
@@ -0,0 +1,189 @@
+//style
+import { ReactComponent as Friends } from "../../assets/icons/Friends.svg";
+import { ReactComponent as Next } from "../../assets/icons/Next.svg";
+import { ReactComponent as Star } from "../../assets/icons/Star.svg";
+import { Body2, Caption1, Heading2 } from "../../style/font";
+import styled from "styled-components";
+//component
+import { useRecoilState, useRecoilValue, useSetRecoilState } from "recoil";
+import {
+  chatRoomListState,
+  friendsState,
+  greenDotState,
+} from "../../recoil/state";
+import { useRef, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+interface ProfileCardProps {
+  userIndex: number;
+  favorite: boolean; // 현재 페이지 상태
+}
+const ProfileCard = (props: ProfileCardProps) => {
+  const navigate = useNavigate();
+  const chatList = useRecoilValue(chatRoomListState);
+  //사용자 제외 친구 list
+  const [users, setUsers] = useRecoilState(friendsState);
+  //즐겨찾기 추가 시 dot
+  const setGreenDot = useSetRecoilState(greenDotState);
+  //favorite 버튼 toggle
+  const [favoriteToggle, setFavoriteToggle] = useState(false);
+  //우클릭 시 메뉴
+  const handleOnContextMenu = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFavoriteToggle(!favoriteToggle);
+  };
+  //즐겨찾기 click
+  const handleFavoriteClick = () => {
+    const copy = [...users];
+    if (copy[props.userIndex].favorite === 0) {
+      copy[props.userIndex] = {
+        ...copy[props.userIndex],
+        favorite: 1,
+      };
+    } else if (copy[props.userIndex].favorite === 1) {
+      copy[props.userIndex] = {
+        ...copy[props.userIndex],
+        favorite: 0,
+      };
+    }
+    //현재 즐겨찾기 상태일 때는 업데이트 점 X
+    if (props.favorite === false) {
+      setGreenDot(true);
+    }
```

Comment:
```suggestion
   const updatedUsers = [...users];
    updatedUsers[props.userIndex] = {
      ...updatedUsers[props.userIndex],
      favorite: updatedUsers[props.userIndex].favorite === 0 ? 1 : 0,
    };
  
    if (!props.favorite) {
      setGreenDot(true);
    }
```
copy라는 변수명은 어떠한걸 복사했는지 잘 알 수 없어서 한번 변수명과 코드를 좀 줄여봤습니다!